calliope/core/model.py

```diff
@@ -321,7 +321,12 @@ def build(self, backend_interface: Literal["pyomo"] = "pyomo") -> None:
         for components in ["variables", "expressions", "constraints", "objectives"]:
             component = components.removesuffix("s")
             for name, dict_ in self.math[components].items():
-                getattr(backend, f"add_{component}")(self._model_data, name, dict_)
+                if dict_.get("active", True):
+                    getattr(backend, f"add_{component}")(self._model_data, name, dict_)
+                else:
+                    logger.debug(
+                        f"({component}, {name}): Component deactivated and therefore not built."
```

Comment:
Is components now the official name for the union of variables, global expressions, constraints and objectives?

Reply:
I don't know 😫 
If I google "components of a linear program" I get these items (decision variables, objective function, constraints, ...), so let's stick with components.